packages/@aws-cdk/aws-appmesh/lib/client-policy.ts

```diff
@@ -0,0 +1,110 @@
+import * as acmpca from '@aws-cdk/aws-acmpca';
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+
+enum CertificateType {
+  ACMPCA = 'acm',
+  FILE = 'file'
+}
+
+/**
+ * Properties of TLS Client Policy
+ */
+export interface ClientPolicyConfig {
+  /**
+   * Represents single Client Policy property
+   */
+  readonly clientPolicy: CfnVirtualNode.ClientPolicyProperty;
+}
+
+/**
+ * Represents the property needed to define a Client Policy
+ */
+export interface ClientPolicyOptions {
+  /**
+   * TLS is enforced on the ports specified here.
+   * If no ports are specified, TLS will be enforced on all the ports.
+   *
+   * @default - none
+   */
+  readonly ports?: number[];
+}
+
+/**
+ * ACM Trust Properties
+ */
+export interface AcmTrustOptions extends ClientPolicyOptions {
+  /**
+   * Contains information for your private certificate authority
+   */
+  readonly certificateAuthorityArns: acmpca.ICertificateAuthority[];
```

Comment:
You missed a comment from my previous review: https://github.com/aws/aws-cdk/pull/11563#discussion_r532950230

```suggestion
  readonly certificateAuthorities: acmpca.ICertificateAuthority[];
```